src/Native/Unix/System.Native/pal_interfaceaddresses.c

```diff
@@ -115,6 +115,15 @@ int32_t SystemNative_EnumerateInterfaceAddresses(IPv4AddressFound onIpv4Found,
             {
                 struct sockaddr_ll* sall = (struct sockaddr_ll*)current->ifa_addr;
 
+                if (sall->sll_halen > sizeof(sall->sll_addr))
+                {
+                    // sockaddr_ll->sll_addr has a maximum capacity of 8 bytes (unsinged char sll_addr[8])
+                    // so if we get a address length greater than that, we truncate it to 8 bytes.
+                    // this is following the kernel docs where they always treat physical addresses with a maximum of 8 bytes
+                    // however in WSL we hit an issue where sll_halen was 16 bytes so the memcpy_s below would fail because it was greater
```

Comment:
Does that mean we lose data on WSL?  Is the remaining data correct?  What gets truncated?

Reply:
The VPN interface we look at did not have any link address (as it is point-to-point) 
Linux ifconfig shows it as all zeros.

Reply:
Ok.